crates/storage/nippy-jar/src/writer.rs

```diff
@@ -43,7 +43,11 @@ pub struct NippyJarWriter<H: NippyJarHeader = ()> {
 
 impl<H: NippyJarHeader> NippyJarWriter<H> {
     /// Creates a [`NippyJarWriter`] from [`NippyJar`].
-    pub fn new(jar: NippyJar<H>) -> Result<Self, NippyJarError> {
+    ///
+    /// If `with_consistency_heal` is set to true, it will self-heal on any inconsistent state. This
+    /// might be undesireable in concurrent situations. If it encounters an issue, it will return an
+    /// error instead.
+    pub fn new(jar: NippyJar<H>, with_consistency_heal: bool) -> Result<Self, NippyJarError> {
```

Comment:
not a fan of plain bool in arguments, would prefer an enum to make it more explicit. Also, do we ever need to skip healing?

Reply:
good point, should be an enum actually